pkg/kancontroller/kancontroller.go

```diff
@@ -46,13 +46,24 @@ const (
 )
 
 func Execute() {
+	execute(nil)
```

Comment:
personally I would move the `config, err = rest.InClusterConfig()` into here and pass the result instead of nil rather than having nested if's in `execute`

Reply:
the other option that we can consider is using the way kubectl or other tools do this.

Reply:
The `k8s.io/cli-runtime/pkg/genericclioptions` will include [options](k8s.io/cli-runtime/pkg/genericclioptions) that are suitable for CLI, but not controller i think.